sycl/plugins/unified_runtime/ur/adapters/hip/enqueue.cpp

```diff
@@ -1054,6 +1064,12 @@ UR_APIEXPORT ur_result_t UR_APICALL urEnqueueMemImageCopy(
     size_t SrcOffset[3] = {SrcByteOffsetX, srcOrigin.y, srcOrigin.z};
     size_t DstOffset[3] = {DstByteOffsetX, dstOrigin.y, dstOrigin.z};
 
+    ur_event_handle_t NewEvent = nullptr;
+    if (phEvent) {
+      NewEvent = ur_event_handle_t_::makeNative(UR_COMMAND_MEM_IMAGE_COPY,
```

Comment:
Should these be wrapped in a `unique_ptr` and `.release()` so that they memory is freed if there's an error or early return in between

Reply:
Change made. Have also added unique ptrs in one or two other spots where needed